File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/valutazioni/ValutazioneTestataxAnno.cs
using AgronicaCoreModelsSTD.baseClass;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.valutazioni
{
    public class Valutazione_TestataxAnno
    {
        public PK primaryKey { get; set; }

        //public string Piva { get; set; }
        //public string Id_Testata { get; set; }
        //public int Anno { get; set; }
        public int Anno_Tipo { get; set; }
        public string Anno_Tipo_Des { get; set; }

        public bool flag_cancellazione { get; set; }

        public Valutazione_TestataxAnno()
        {
            this.flag_cancellazione = false;
        }

        public Valutazione_TestataxAnno(PK primaryKey)
        {
            this.primaryKey = primaryKey;
            this.flag_cancellazione = false;
        }

        public class PK
        {
            public int Anno { get; set; }
            public Valutazione_Testata.PK valutazioneTestataPK { get; set; }


            public PK(int anno, Valutazione_Testata.PK valutazioneTestataPK)
            {
                this.Anno = anno;
                this.valutazioneTestataPK = valutazioneTestataPK;
            }

            public PK()
            {

            }

        }
    }

}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/valutazioni/ValutazioneTestataxAnno.ts
import { PKValutazioneTestata } from './ValutazioneTestata';
import { BaseCodeDescrStr } from '../baseClass/baseCodeDescrStr';

export type PKValutazioneTestataxAnno = typeof ValutazioneTestataxAnno.PK.prototype;
export class ValutazioneTestataxAnno {

    static PK = class {
        Anno: number;
        valutazioneTestataPK: PKValutazioneTestata;

        constructor(anno: number, valutazioneTestataPK: PKValutazioneTestata) {
            this.Anno = anno;
            this.valutazioneTestataPK = valutazioneTestataPK;
        }
    };

    primaryKey: PKValutazioneTestataxAnno;

    Anno_Tipo: number;
    Anno_Tipo_Des: string;

    flag_cancellazione: boolean;

    constructor(primaryKey: PKValutazioneTestataxAnno) {
        this.primaryKey = primaryKey;
        this.flag_cancellazione = false;
    }

}
